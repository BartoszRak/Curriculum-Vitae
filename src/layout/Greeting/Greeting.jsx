import React from 'react'
import { FormattedMessage } from 'react-intl'
import { withStyles, Typography, Avatar } from '@material-ui/core'
import PropTypes from 'prop-types'


import avatarUrl from '~assets/images/instagram-photo.jpg'

import style from './Greeting.style'

export function Greeting({ classes }) {
  return (
    <header
      className={classes.root}
    >
      <Avatar alt="Author instagram picture" className={classes.avatar} src={avatarUrl} />
      <Typography className={classes.title} color="secondary" variant="h1">
        Bartosz Rak
      </Typography>
      <Typography className={classes.subtitle} color="secondary" variant="h2">
        <FormattedMessage id="layout.greeting.subtitle" />
        <span aria-hidden>üòÅ</span>
      </Typography>
      <Typography className={classes.subsubtitle} color="secondary" variant="h3">
        <FormattedMessage id="layout.greeting.subsubtitle" />
      </Typography>
    </header>
  )
}

Greeting.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(style)(Greeting)
