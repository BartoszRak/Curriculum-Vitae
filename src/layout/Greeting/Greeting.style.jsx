export default () => ({
  root: {
    alignItems: 'center',
    backgroundAttachment: 'fixed',
    display: 'flex',
    flexDirection: 'column',
    height: '100vh',
    justifyContent: 'center',
  },
})
