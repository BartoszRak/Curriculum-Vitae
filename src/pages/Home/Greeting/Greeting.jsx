import React from 'react'
import { FormattedMessage } from 'react-intl'
import { Typography, Avatar, Container } from '@material-ui/core'
import PropTypes from 'prop-types'

import avatarUrl from '~assets/images/instagram-photo2.jpg'

import useStyle from './Greeting.style'

export function Greeting({ classes: overridingClasses }) {
  const classes = { ...useStyle(), ...overridingClasses }
  return (
    <section className={classes.section}>
      <Container classes={{ root: classes.root }} id="greeting">
        <Avatar
          alt="Author instagram picture"
          className={classes.avatar}
          src={avatarUrl}
        />
        <Typography
          className={classes.title}
          color="secondary"
          data-testid="greeting-title"
          variant="h1"
        >
          Bartosz Rak
        </Typography>
        <Typography
          className={classes.subtitle}
          color="secondary"
          data-testid="greeting-subtitle"
          variant="h2"
        >
          <FormattedMessage id="pages.home.greeting.subtitle" />
          <span aria-hidden>üòÅ</span>
        </Typography>
        <Typography
          className={classes.subsubtitle}
          color="secondary"
          data-testid="greeting-subsubtitle"
          variant="h3"
        >
          <FormattedMessage id="pages.home.greeting.subsubtitle" />
        </Typography>
      </Container>
    </section>
  )
}

Greeting.propTypes = {
  classes: PropTypes.object,
}

export default Greeting
