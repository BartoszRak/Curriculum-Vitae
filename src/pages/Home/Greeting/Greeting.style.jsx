import { makeStyles } from '@material-ui/core'

export default makeStyles(({ breakpoints, palette, spacing }) => ({
  avatar: {
    border: `2px solid ${palette.secondary.main}`,
    height: 250,
    marginBottom: 60,
    width: 250,
  },
  root: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    width: '100%',
  },
  section: {},
  subsubtitle: {
    fontWeight: 'lighter',
    marginTop: spacing(0.5),
    textAlign: 'center',
    textShadow: '0 0 5px rgba(0, 0, 0, 0.2)',
    [breakpoints.down('sm')]: {
      fontSize: '1.2rem',
    },
  },
  subtitle: {
    textAlign: 'center',
    textShadow: '0 0 5px rgba(0, 0, 0, 0.2)',
    [breakpoints.down('sm')]: {
      fontSize: '1.6rem',
    },
  },
  title: {
    textAlign: 'center',
    textShadow: '0 0 5px rgba(0, 0, 0, 0.2)',
    [breakpoints.down('sm')]: {
      fontSize: '2.5rem',
    },
  },
}))
