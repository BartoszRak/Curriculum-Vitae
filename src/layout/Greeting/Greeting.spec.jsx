import React from 'react'
import { shallow } from 'enzyme'

import { Greeting } from './Greeting'

const initComponent = overrides => {
  const mockProps = {
    classes: {},
  }
  const mockMethods = {}
  const wrapper = shallow(<Greeting {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('global: Greeting', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
  it('should render as expected', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
