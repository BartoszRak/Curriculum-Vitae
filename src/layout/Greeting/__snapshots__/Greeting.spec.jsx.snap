// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Greeting should render as expected 1`] = `
<header>
  <AvatarMock
    alt="Author instagram picture"
    src="instagram-photo.jpg"
  />
  <TypographyMock
    color="secondary"
    variant="h1"
  >
    Bartosz Rak
  </TypographyMock>
  <TypographyMock
    color="secondary"
    variant="h2"
  >
    <FormattedMessage
      id="layout.greeting.subtitle"
      values={Object {}}
    />
    <span
      aria-hidden={true}
    >
      😁
    </span>
  </TypographyMock>
  <TypographyMock
    color="secondary"
    variant="h3"
  >
    <FormattedMessage
      id="layout.greeting.subsubtitle"
      values={Object {}}
    />
  </TypographyMock>
</header>
`;
